sigstore/transparency.py

```diff
@@ -131,7 +131,13 @@ def __post_init__(self) -> None:
         """
         Invariant preservation.
         """
-        if self.inclusion_proof is None and self.inclusion_promise is None:
+
+        # An inclusion proof isn't considered present unless its checkpoint
+        # is also present.
+        has_incluson_proof = (
+            self.inclusion_proof is not None and self.inclusion_proof.checkpoint
```

Comment:
Do we want to log something when there's an inclusion proof but no checkpoint? Seems like it might be a bit confusing that we silently ignore the inclusion proof.

Reply:
Hmm, I need to think about where it makes sense to add this check + log: it might make more sense in `from_bundle`, since that's really the only place we'd expect user error to cause a mismatch here.

Reply:
https://github.com/sigstore/sigstore-python/pull/723/commits/2ffcf089f34062db982358ee4fde32dda1064147